connectors/src/connectors/slack/lib/workspace_limits.ts

```diff
@@ -212,19 +200,35 @@ export async function isActiveMemberOfWorkspace(
   return workspaceActiveMemberEmails.includes(slackUserEmail);
 }
 
-function isBotAllowed(slackUserInfo: SlackUserInfo) {
+async function isBotAllowed(
+  connector: ConnectorResource,
+  slackUserInfo: SlackUserInfo
+) {
   const displayName = slackUserInfo.display_name ?? "";
   const realName = slackUserInfo.real_name ?? "";
 
-  const isWhitelistedBotName =
-    WHITELISTED_BOT_NAME.includes(displayName) ||
-    WHITELISTED_BOT_NAME.includes(realName);
-
-  if (!isWhitelistedBotName) {
-    logger.info({ user: slackUserInfo }, "Ignoring bot message");
+  const names = [displayName, realName].filter((name) => name.length > 0);
+
+  // Whitelisting a bot will accept any message from this bot.
+  // This means that even a non verified user of a given Slack workspace who can trigger a bot
+  // that talks to our bot (@dust) will be able to use the Dust bot.
+  // Make sure to be explicit about this with users as you whitelist a new bot.
+  // Example: non-verfied-user -> @AnyWhitelistedBot -> @dust -> Dust answers with potentially private information.
+  const whitelist = await SlackBotWhitelistModel.findOne({
```

Comment:
This should be behind the slackConfigurationResource API no?

Reply:
➕ Any reason not to gate this behind the `SlackConfigurationResource`?